File: examples/governance/identity_domain/lib/identity_domain/stakeholder/commands/deactivate_stakeholder.rb
module IdentityDomain
  class Stakeholder
    module Commands
      class DeactivateStakeholder
        include Hecks::Command
        emits "DeactivatedStakeholder"

        attr_reader :stakeholder_id

        def initialize(stakeholder_id: nil)
          @stakeholder_id = stakeholder_id
        end

        def call
          existing = repository.find(stakeholder_id)
          if existing
            unless existing.status == "active"
              raise Hecks::Error, "Cannot DeactivateStakeholder: status must be 'active', got '#{existing.status}'"
            end
            Stakeholder.new(
              id: existing.id,
              name: existing.name,
              email: existing.email,
              role: existing.role,
              team: existing.team,
              status: "deactivated"
            )
          else
            raise Hecks::Error, "Stakeholder not found: #{stakeholder_id}"
          end
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/deactivate_stakeholder.go
package domain

import (
	"time"
	"fmt"
)

type DeactivateStakeholder struct {
	StakeholderId string `json:"stakeholder_id"`
}

func (c DeactivateStakeholder) CommandName() string { return "DeactivateStakeholder" }

func (c DeactivateStakeholder) Execute(repo StakeholderRepository) (*Stakeholder, *DeactivatedStakeholder, error) {
	existing, err := repo.Find(c.StakeholderId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("Stakeholder not found: %s", c.StakeholderId)
	}
	if existing.Status != "active" {
		return nil, nil, fmt.Errorf("cannot DeactivateStakeholder: Stakeholder is in %s state", existing.Status)
	}
	existing.Status = "deactivated"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := DeactivatedStakeholder{
		AggregateID: existing.ID,
		StakeholderId: c.StakeholderId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}
